File: Task/Sleeping-Beauty-problem/Go/sleeping-beauty-problem.go
package main

import (
    "fmt"
    "math/rand"
    "rcu"
    "time"
)

func sleepingBeauty(reps int) float64 {
    wakings := 0
    heads := 0
    for i := 0; i < reps; i++ {
        coin := rand.Intn(2) // heads = 0, tails = 1 say
        wakings++
        if coin == 0 {
            heads++
        } else {
            wakings++
        }
    }
    fmt.Printf("Wakings over %s repetitions = %s\n", rcu.Commatize(reps), rcu.Commatize(wakings))
    return float64(heads) / float64(wakings) * 100
}

func main() {
    rand.Seed(time.Now().UnixNano())
    pc := sleepingBeauty(1e6)
    fmt.Printf("Percentage probability of heads on waking = %f%%\n", pc)
}

File: Task/Sleeping-Beauty-problem/Lua/sleeping-beauty-problem.lua
do  -- sleeping beauty problem - translated from the Wren sample, via Pluto

    local function sleepingBeauty( reps )
        local wakings, heads = 0, 0
        for _ = 1, reps do
            wakings = wakings + 1
            if math.random() < 0.5 then
                heads   = heads   + 1 -- [0..0.5) = heads
            else
                wakings = wakings + 1 -- [0.5..1.0) = tails
            end
        end
        print( "Wakings over " .. reps .. " repetitions = " .. wakings )
        return ( heads / wakings ) * 100
    end

    print( "Percentage probability of heads on waking = " .. sleepingBeauty( 1000000 ) )
end
